rsa/prime.py

```diff
@@ -171,6 +171,86 @@ def getprime(nbits):
 
             # Retry if not prime
 
+small_primes=(3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61,
+              67, 71, 73, 79, 83, 89, 97)
+def prime_sieve(start,end):
+    """for numbers in range(start,end) sieves out composites using trial
+    division returning potential primes
+
+    >>> sieve=prime_sieve(100,120)
+    >>> next(sieve)
+    101
+    >>> [x for x in sieve]
+    [103, 107, 109, 113]
+    """
+
+    #handle small numbers
+    if start<=small_primes[-1]:
+        if start<=2:yield 2
+        for p in small_primes:
+            if p<=start:yield p
+    start|=1#make start odd
+    #We use an offset when doing the trial divisions. It is much smaller than
+    #the full number. This makes the modulo operations fast. When yielding a
+    #candidate we add the start and offset to get the candidate value.
+    residues=tuple((-start%p,p) for p in small_primes)
```

Comment:
You have a commenting style that explains part of what is happening above the code, and another part below the code. I like that you explain what's going on, but the style is inconsistent with the rest of the code. Furthermore, I find it hard to understand until I read past the code and see that there is more explanation there. Please just explain everything above the code.